modules/backend/clinic/staff/doctor/DoctorFactory.cpp

```diff
@@ -0,0 +1,17 @@
+#include "DoctorFactory.hpp"
+
+namespace clinic
+{
+namespace staff
+{
+namespace doctor
+{
+
+DoctorPtr createDoctor(const common::PersonalData& personal_data, const common::Specialization specialization)
+{
+    return std::make_unique<Doctor>(personal_data, specialization);
+}
```

Comment:
I wonder if it makes sense to create another layer with create functions in our new way of work, when there are no static memory concepts.

One could just simply call std::make_unique<Doctor>(personal_data, specialization);.

It forces us to create DoctorFactory class just for that purpose

Reply:
1. It does not force us for factory 
2. The reason why I introduced this function is as follows:
```
std::unique_ptr<iDoctor> temp_doc{...};

doc_collection.addDoctor(std::move(temp_doc)); <--- wanted to get rid of such situation when someone forgets about std::move here. 
```

Reply:
Ok, it there is any reason behind this, than I won't argue with it.
Most likely, you are right here. After some thought I realized, that it might lead to cleaner code, especially as we using unique_ptr.
Just BTW, the problem above would most likely be detected by compiler, as unique_ptr can't be copied.